app/components/FreeDeckFeed/FreeDeckFeed.tsx

```
@@ -1,35 +1,62 @@
 import { DECK_LIMIT } from "@/app/constants/decks";
 import { getFreeDecks } from "@/app/queries/home";
 import { useInfiniteQuery } from "@tanstack/react-query";
-import React, { useMemo } from "react";
+import React, { useMemo, useState } from "react";
 
+import ErrorDeck from "../ErrorDeck/ErrorDeck";
 import { HomeFeedDeckCard } from "../HomeFeedDeckCard/HomeFeedDeckCard";
 import LoadMore from "../LoadMore/LoadMore";
 import NoDeck from "../NoDecks/NoDeck";
 
 function FreeDeckFeed() {
-  const { data, fetchNextPage, hasNextPage, isFetching } = useInfiniteQuery({
-    queryKey: ["free-decks"],
-    queryFn: ({ pageParam }) => getFreeDecks({ pageParam }),
-    initialPageParam: 1,
-    getNextPageParam: (lastPage, allPages) => {
-      const totalCount = lastPage?.[0]?.total_count;
-      const totalPages = totalCount
-        ? Math.ceil(totalCount / DECK_LIMIT)
-        : allPages.length;
-      if (totalPages === allPages.length) {
-        return undefined;
-      }
-      return allPages.length + 1;
-    },
-  });
+  const [hasError, setHasError] = useState(false);
+  const { data, fetchNextPage, hasNextPage, isFetching, refetch } =
+    useInfiniteQuery({
+      queryKey: ["free-decks"],
+      queryFn: ({ pageParam }) => {
+        try {
+          if (hasError) setHasError(false);
+          return getFreeDecks({ pageParam });
+        } catch {
+          setHasError(true);
+          return []; // Return empty array to prevent cascading error
```

Will this result in all of the data from the page being erased when it hits this point? Or will existing data persist?

---

Oh based on https://github.com/gator-labs/chomp/pull/1020#issuecomment-2708160938 I do think it clears all data on failure. Is it possible to swallow the error and keep the existing data on screen?

---

Correct, it will clear all data on failure. I was considering an edge case related to Replay—if a user opens a deck from the deck carousel, completes it, and then returns home, an error at that moment could result in invalidated cached data. If the error is swallowed, the answered deck might still appear incorrectly.

---

What would be the root cause of an error you are imagining that causes an error after answering a deck and returning home.

I'm thinking of the case where Cloudflare security check times out after leaving a tab open for a long time. Given these changes, I would expect users to always see an error in that case. I think this will happen a lot.

---

Let's move forward with reloading page when error occurs.